clients/client-python/tests/integration/base_hadoop_env.py

```diff
@@ -0,0 +1,89 @@
+"""
+Copyright 2024 Datastrato Pvt Ltd.
+This software is licensed under the Apache License version 2.
+"""
+
+import logging
+import os
+import shutil
+import subprocess
+import tarfile
+
+import requests
+
+from gravitino.exceptions.gravitino_runtime_exception import GravitinoRuntimeException
+
+logger = logging.getLogger(__name__)
+
+HADOOP_VERSION = "2.7.3"
+HADOOP_PACK_NAME = f"hadoop-{HADOOP_VERSION}.tar.gz"
+HADOOP_DIR_NAME = f"hadoop-{HADOOP_VERSION}"
+HADOOP_DOWNLOAD_URL = f"https://archive.apache.org/dist/hadoop/core/hadoop-{HADOOP_VERSION}/{HADOOP_PACK_NAME}"
+LOCAL_BASE_DIR = "/tmp/gravitino"
+LOCAL_HADOOP_DIR = f"{LOCAL_BASE_DIR}/python_its/hadoop"
+
+
+class BaseHadoopEnvironment:
+
+    @classmethod
+    def init_hadoop_env(cls):
+        # download hadoop pack and unzip
+        if not os.path.exists(LOCAL_HADOOP_DIR):
+            os.makedirs(LOCAL_HADOOP_DIR)
+        cls._download_and_unzip_hadoop_pack()
+        # configure hadoop env
+        cls._configure_hadoop_environment()
+
+    @classmethod
+    def clear_hadoop_env(cls):
+        try:
+            shutil.rmtree(LOCAL_BASE_DIR)
+        except OSError as e:
+            logger.warning("Failed to delete directory '%s': %s", LOCAL_BASE_DIR, e)
+
+    @classmethod
+    def _download_and_unzip_hadoop_pack(cls):
+        logger.info("Download and unzip hadoop pack from %s.", HADOOP_DOWNLOAD_URL)
+        local_tar_path = f"{LOCAL_HADOOP_DIR}/{HADOOP_PACK_NAME}"
```

Comment:
I would suggest to download to a temporary unique folder, after successfully downloaded and unzipped, then move to the destination you wanted, that will increase the robustness.